### PR TITLE
Allow Icons as rightElements in inputs, Fixes #1803

### LINKED ISSUE #1803: NumberInput.rightElement prop unusable
I wanted to use the documented `NumberInput.rightElement` prop to display a right-aligned icon (in my case, a percent sign as my input was holding a rate), but the element did not render as expected. Using a simple text string also did not render properly.

It appears as though this element was getting a zero default width. Attempting to set a width just made things worse. (To replicate, try to do something useful with this prop 😄)

### PR DESCRIPTION
**Pull request by febbraiod**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [NR] Updated doc comments / prop-types, or determined not required.
- [NR] Reviewed and tested on Mobile, or determined not required.
- [NR] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: cmp/input/HoistInput.scss [hunk 1]
```diff
@@ -12,3 +12,9 @@
     }
   }
 }
+
+.icon-as-right-element {
```
Comment (febbraiod) [inline cmp/input/HoistInput.scss]: Not in love with this name by any stretch

**Comment by lbwexler** [comment]
Thanks Don!  Tom -- can you review and merge?

**Comment by amcclain** [comment]
Hi - I looked at this for a while on the plane as I had been curious what was going on.

See https://github.com/xh/hoist-react/commit/7251f2152c5f0b7eb06483dfb0ebef3ebc8b4c71 for an alternate approach - it led me down a bit of a rabbit hole with how elements are sized and ultimately this bit of code in the BP input that sets up the padding added directly to the input element - https://github.com/palantir/blueprint/blob/0c7c2090fcf0e3e65ee5ab434d623df53912e9ee/packages/core/src/components/forms/inputGroup.tsx#L158

**Comment by TomTirapani** [comment]
Thanks for this Don! Looks good, and solves the issue. :)

However, I think it would be better to take Anselm's alternative mentioned above - the change is more minimal, doesn't require our inputs to check the type of element passed and gets to the root of the issue w.r.t. BP's sizing calculations (and supports any width for icon, should we want to).

I'll close this in favour of that PR, make a couple of tweaks there and merge.

**Pull request closed without merging**